diff --git a/src/config.c b/src/config.c
index 5ba0f5c..9aac13f 100644
--- a/src/config.c
+++ b/src/config.c
@@ -424,6 +424,8 @@ static bool HandleIniConfig(int section, const char *key, char *value) {
           g_config.extended_aspect_ratio = (h * 16 / 10 - 256) / 2;
         else if (strcmp(s, "18:9") == 0)
           g_config.extended_aspect_ratio = (h * 18 / 9 - 256) / 2;
+        else if (strcmp(s, "3:2") == 0)
+          g_config.extended_aspect_ratio = (h * 3 / 2 - 256) / 2;
         else if (strcmp(s, "4:3") == 0)
           g_config.extended_aspect_ratio = 0;
         else if (strcmp(s, "unchanged_sprites") == 0)
